pkg/cloudprovider/aws/instance.go

```diff
@@ -35,33 +35,46 @@ type InstanceProvider struct {
 }
 
 // Create an instance given the constraints.
+// instanceTypeOptions should be sorted by priority for spot capacity type.
```

Comment:
What if Spot isn't being used?

Reply:
If spot isn't used, then we don't want to prioritize the instance types (since we're using lowest-price), so it doesn't really matter if they're passed in sorted or not.  I can make that more explicit on the func comment.